rust/noosphere-core/src/authority/author.rs

```diff
@@ -103,6 +107,34 @@ where
 
         Ok(Access::ReadOnly)
     }
+
+    /// A helper for creating and signing a new UCAN token for publishing.
+    pub async fn create_publish_token(
```

Comment:
This seems like a nice function to have along with the NS/DHT crate, though designed to be non-web, wonder if we can still expose some data types like `NSRecord` or the token gen utilities, or maybe these are core capabilities.